downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/ProtegeExporter.java

```diff
@@ -0,0 +1,414 @@
+				{
+					logger.info("Shutting down thread pool. IF this program was terminated prematurely, there might still be orphan Perl processe runing."
+								+ " Check to see if any are still running, and kill them if necssary.");
+					pool.shutdownNow();
+				}));
+				// Filtering is in effect IF the set has something in it.
+				final boolean idFilterInEffect = !this.pathwayIdsToProcess.isEmpty();
+				if (idFilterInEffect)
+				{
+					logger.info("Filtering by ID is in effect. ONLY the following pathways will be exported to protege: {}", this.pathwayIdsToProcess.toString());
+				}
+				final boolean speciesFilterInEffect = !this.speciesToProcess.isEmpty();
+				if (speciesFilterInEffect)
+				{
+					logger.info("Filtering by species is in effect. ONLY pathways with the following species will be exported to protege: {}", this.speciesToProcess.toString());
+				}
+				LocalDateTime overallStart = LocalDateTime.now();
+				// submit jobs to pool:
+				pool.submit(() ->
+				{
+					// parallelStream should use the degree of parallelism set on the "pool" object.
+					pathways.parallelStream().forEach(pathway -> 
+					{
+						boolean processPathway = shouldPathwayBeProcessed(idFilterInEffect, speciesFilterInEffect, pathway);
+						
+						if (processPathway)
+						{
+							logger.info("Running protegeexport script for Pathway: {}", pathway.toString());
+							// include a normalized display_name in the output to make it easier to identify the contents of an archive.
+							String normalizedDisplayName = pathway.getDisplayName().toLowerCase().replace(" ", "_").replace("-", "_").replace("(", "").replace(")", "");
+							String fileName = "Reactome_pathway_" + pathway.getDBID() + "_" + normalizedDisplayName;
+							ProcessBuilder processBuilder = createProcessBuilder(pathway.getDBID().toString(), fileName);
+							Process process = null;
+							try
+							{
+								LocalDateTime exportStart = LocalDateTime.now();
+								process = processBuilder.start();
+								// Now, wait for protege export to complete.
+								int exitCode = process.waitFor();
+								LocalDateTime exportEnd = LocalDateTime.now();
+								logger.info("Finished generating protege files for {}; Elapsed time: {}; Exit code is {}", pathway.toString(), Duration.between(exportStart, exportEnd), exitCode);
+								if (exitCode == 0)
+								{
+									Files.createDirectories(Paths.get("/tmp/protege_files"));
+									// Move the file to the protege_files, overwrite existing.
+									Files.move(Paths.get("/tmp/"+fileName+".tar.gz"), Paths.get(PROTEGE_FILES_DIR+fileName+".tar.gz"), StandardCopyOption.REPLACE_EXISTING);
+								}
+								else
+								{
+									// Should a non-zero exit throw an exception and interrupt the whole process? I am leaning to "no" - if a *single* pathway fails to export,
+									// I think I'd want the rest of them to continue, and then we can re-run for just the failed pathway.
```

Comment:
I agree, if any fail we can just re-run the script with a filter set to just try and generate the failed ones